src/Temperature-Accretech_UFseries/main.py

```diff
@@ -113,26 +119,46 @@ def connect(self) -> None:
         #     self.device_communication[self.instance_key] = self.port
 
         self.prober = accretech_uf.AccretechProber(self.port)
-
         self.prober.set_verbose(self.verbosemode)
 
+        self.unique_identifier = "Driver_Accretech_UFseries_" + self.port_str
+        if not self.unique_identifier in self.device_communication:
+            self.device_communication[self.unique_identifier] = None
+
     def disconnect(self) -> None:
         """Disconnect from the device."""
-        del self.prober  # this makes sure the event mechanism is disabled
+        if self.unique_identifier in self.device_communication:
+            self.prober.unregister_srq_event()  # this makes sure the event mechanism is disabled
```

Comment:
The unregister is called only once for a unique identifier. But in connect() above the register is called for every instance (in the `__init__`  of AccretechProber) even if the port is the same. This asymmetry does not make any problems?

Reply:
There was no problem because registering again does not registering it twice, but removing it leads to the fact that it cannot be removed again.
I will not change it as this would have to be tested otherwise.